This fails to build on some platforms only (ppc64), for mysterious reasons.
It should fail everywhere as it's wrong (links to the *64 symbols).

diff --git a/vendor/rustix/src/backend/libc/fs/types.rs b/vendor/rustix/src/backend/libc/fs/types.rs
index 8d8ec08..cac5816 100644
--- a/vendor/rustix/src/backend/libc/fs/types.rs
+++ b/vendor/rustix/src/backend/libc/fs/types.rs
@@ -863,7 +863,6 @@ pub enum FlockOperation {
 /// [`fstat`]: crate::fs::fstat
 #[cfg(not(any(
     target_os = "android",
-    target_os = "linux",
     target_os = "emscripten",
     target_os = "l4re",
 )))]
@@ -875,7 +874,7 @@ pub type Stat = c::stat;
 /// [`fstat`]: crate::fs::fstat
 #[cfg(any(
     all(
-        any(target_os = "android", target_os = "linux"),
+        any(target_os = "android"),
         target_pointer_width = "64",
     ),
     target_os = "emscripten",
diff --git a/vendor/rustix/src/backend/libc/offset.rs b/vendor/rustix/src/backend/libc/offset.rs
index 8aae9d0..19fc54e 100644
--- a/vendor/rustix/src/backend/libc/offset.rs
+++ b/vendor/rustix/src/backend/libc/offset.rs
@@ -8,7 +8,6 @@ use super::c;
     target_os = "android",
     target_os = "emscripten",
     target_os = "l4re",
-    target_os = "linux",
 )))]
 #[cfg(feature = "fs")]
 pub(super) use c::{
@@ -20,7 +19,6 @@ pub(super) use c::{
     target_os = "android",
     target_os = "emscripten",
     target_os = "l4re",
-    target_os = "linux",
 ))]
 #[cfg(feature = "fs")]
 pub(super) use c::{
